### PR TITLE
Fix String.Extra doc comments (use '-->')

### PR DESCRIPTION
**Pull request by Janiczek**
This makes the doc tests in `String.Extra` runnable.
Unfortunately it also shows that some of them are failing (see below).

I think the failures are a mix of code being wrong and tests being wrong. Some hammock time might be needed to figure out what the expected behaviour of those functions is supposed to be.

@gampleman Just for transparency, I am not planning to fix these - I just tried the `==` ➡️ `-->` change in case it was non-problematic. Ultimately it wasn't, but I hope my commit here will save you or anybody picking up this work, _some_ of the work.

```
↓ VerifyExamples.String.Extra.SoftWrap2
✗ #softWrap:

    softWrap 7 "My very long text"
    --> "My very\nlong text"

    "My very\nlong text"
    ╷
    │ Expect.equal
    ╵
    "My \nvery \nlong \ntext"


↓ VerifyExamples.String.Extra.SoftWrapWith2
✗ #softWrapWith:

    softWrapWith 7 "..." "My very long text"
    --> "My very...long text"

    "My very...long text"
    ╷
    │ Expect.equal
    ╵
    "My ...very ...long ...text"


↓ VerifyExamples.String.Extra.Unindent1
✗ #unindent:

    unindent "  Hello\n    World "
    --> "Hello\n  World"

    "Hello\n  World"
    ╷
    │ Expect.equal
    ╵
    "Hello\n  World "


↓ VerifyExamples.String.Extra.Wrap1
✗ #wrap:

    wrap 7 "My very long text"
    --> "My very\nlong te\nxt"

    "My very\nlong te\nxt"
    ╷
    │ Expect.equal
    ╵
    "My very\n long t\next"


↓ VerifyExamples.String.Extra.WrapWith1
✗ #wrapWith:

    wrapWith 7 "\n" "My very long text"
    --> "My very\nlong text"

    "My very\nlong text"
    ╷
    │ Expect.equal
    ╵
    "My very\n long t\next"



TEST RUN FAILED

Duration: 514 ms
Passed:   1040
Failed:   5
```

### CODE REVIEW

File: src/String/Extra.elm [hunk 1]
```diff
@@ -306,9 +306,9 @@ quote string =
 
 {-| Remove quotes that surround a string.
 
-    unquote "\"foo\"" == "foo"
+    unquote "\"foo\"" --> "foo"
 
-    unquote "\"foo\"bar\""
+    unquote "\"foo\"bar\"" --> "foo\"bar"
```
Comment (Janiczek) [inline src/String/Extra.elm:311]: The "expected" part was missing here.

File: src/String/Extra.elm [hunk 2]
```diff
@@ -650,8 +655,9 @@ toSentenceHelper lastPart sentence list =
 
 {-| Remove all HTML tags from the string, preserving the text inside them.
 
-    stripTags "a <a href=\"#\">link</a>" == "a link"
-    stripTags "<script>alert('hello world!')</script> == "alert('hello world!')"
+    stripTags "a <a href=\"#\">link</a>" --> "a link"
+
+    stripTags "<script>alert('hello world!')</script>" --> "alert('hello world!')"
```
Comment (Janiczek) [inline src/String/Extra.elm:660]: An ending `"` was missing here.

**Comment by miniBill** [comment]
@Janiczek tests are failing, can you fix them?

**Comment by Janiczek** [comment]
@miniBill Sorry, no I won't:
Screenshot 2023-11-15 at 15 54 46

Maybe best to close this PR? It served a purpose of finding out what the situation is with the `-->` doc comments: it's going to be a non-trivial amount of work to fix those.

**Comment by miniBill** [comment]
Ah, sorry, I had missed that. I think it makes sense to keep it open! Doc tests should not be broken!!!

**Comment by gampleman** [comment]
I've added a commit that fixes the tests, so I think we can now merge this, right?

**Pull request merged**